app/scripts/views/elements/token-code-input.js

```diff
@@ -0,0 +1,33 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const AuthErrors = require('../../lib/auth-errors');
+const textInput = require('./text-input');
+const Vat = require('../../lib/vat');
+
+const element = Object.create(textInput);
+
+element.match = function ($el) {
+  return $el.attr('type') === 'number' && $el.hasClass('token-code');
+};
+
+element.val = function (val) {
+  if (arguments.length === 1) {
+    return this.__val(val.replace(/[- ]*/g, ''));
```

Comment:
Fwiw, the `*` seems unnecessary in this because you're already replacing with the `/g` switch.

(and even without that it seems like `+` would be the better choice anyway, because you wouldn't want to match against zero occurrences?)